<!doctype html>
<html class="no-js" lang="en" data-content_root="../">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" />

    <!-- Generated with Sphinx 7.4.7 and Furo 2024.08.06 -->
        <title>Applying CellPhenoX to Ulcerative Colitis Fibroblasts (single-cell transcriptomics) - CellPhenoX 1.0.3 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=8f2a1f02" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?v=354aac6f" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/nbsphinx-code-cells.css?v=2aa19091" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?v=302659d7" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --color-brand-primary: #C25757;
  --color-brand-content: #C25757;
  --color-api-pre-name: #C25757;
  --color-api-name: #C25757;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  --color-brand-primary: #317EC2;
  --color-brand-content: #317EC2;
  --color-api-pre-name: #317EC2;
  --color-api-name: #317EC2;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  --color-brand-primary: #317EC2;
  --color-brand-content: #317EC2;
  --color-api-pre-name: #317EC2;
  --color-api-name: #317EC2;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>

<a class="skip-to-content muted-link" href="#furo-main-content">Skip to content</a>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">CellPhenoX 1.0.3 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
          <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../_static/pycpx.svg" alt="Logo"/>
  </div>
  
  <span class="sidebar-brand-text">CellPhenoX 1.0.3 documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../README.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html#tutorials">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html#api">API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html#license">License</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html#code-of-conduct">Code of Conduct</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html#contributing">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html#issues">Issues</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html#citation">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html#contact">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../CONTRIBUTING.html">Contributing to pyCellPhenoX</a></li>
<li class="toctree-l1"><a class="reference internal" href="../CODE_OF_CONDUCT.html">Code of Conduct - pyCellPhenoX</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../walkthrough.html">Walkthroughs</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of Walkthroughs</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="apply_cellphenoX_inflamed_uc_fibroblast.html">Applying CellPhenoX to Ulcerative Colitis Fibroblasts (single-cell transcriptomics)</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../modules.html">pyCellPhenoX</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle navigation of pyCellPhenoX</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2 has-children"><a class="reference internal" href="../pyCellPhenoX.html">Core functions</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" role="switch" type="checkbox"/><label for="toctree-checkbox-3"><div class="visually-hidden">Toggle navigation of Core functions</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="../pyCellPhenoX.operations.html">Functions</a></li>
<li class="toctree-l3"><a class="reference internal" href="../pyCellPhenoX.utils.html">pyCellPhenoX utilities</a></li>
</ul>
</li>
</ul>
</li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="view-this-page">
  <a class="muted-link" href="../_sources/vignettes/test_vignette.ipynb.txt" title="View this page">
    <svg><use href="#svg-eye"></use></svg>
    <span class="visually-hidden">View this page</span>
  </a>
</div>
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
              <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main" id="furo-main-content">
          <section id="Applying-CellPhenoX-to-Ulcerative-Colitis-Fibroblasts-(single-cell-transcriptomics)">
<h1>Applying CellPhenoX to Ulcerative Colitis Fibroblasts (single-cell transcriptomics)<a class="headerlink" href="#Applying-CellPhenoX-to-Ulcerative-Colitis-Fibroblasts-(single-cell-transcriptomics)" title="Link to this heading">¶</a></h1>
<section id="Goal:-Identify-subtle,-fibroblast-specific-phenotypic-changes-associated-with-differing-levels-of-inflammation">
<h2>Goal: Identify subtle, fibroblast-specific phenotypic changes associated with differing levels of inflammation<a class="headerlink" href="#Goal:-Identify-subtle,-fibroblast-specific-phenotypic-changes-associated-with-differing-levels-of-inflammation" title="Link to this heading">¶</a></h2>
<p>This tutorial explores the utility of CellPhenoX within a specific cell type. Using the single-cell transcriptomics data from <a class="reference external" href="https://doi.org/10.1016/j.cell.2019.06.029">Smillie et al.</a>. We present the step-by-step workflow for running CellPhenoX to identify specific fibroblasts with predictable capability in distinguishing inflamed from non-inflamed colon.</p>
</section>
<section id="Importing-Dependencies">
<h2>Importing Dependencies<a class="headerlink" href="#Importing-Dependencies" title="Link to this heading">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>#import pyCellPhenoX
import sys
sys.path.append(&#39;/Users/zhanglab_mac2/Documents/GitHub/pyCellPhenoX&#39;)
import pyCellPhenoX.CellPhenoX
import pyCellPhenoX.preprocessing
from pyCellPhenoX.neighborhoodAbundanceMatrix import neighborhoodAbundanceMatrix
import pyCellPhenoX.nonnegativeMatrixFactorization
import pyCellPhenoX.principalComponentAnalysis

import pandas as pd
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
</pre></div></div>
</div>
</section>
<section id="Step-1:-Import-data">
<h2>Step 1: Import data<a class="headerlink" href="#Step-1:-Import-data" title="Link to this heading">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># paths to expression data and meta data files
expression_file = &quot;/Users/zhanglab_mac2/Documents/GitHub/pyCellPhenoX/vignettes/example_UC_data/fibroblast_exp.csv&quot;
meta_file = &quot;/Users/zhanglab_mac2/Documents/GitHub/pyCellPhenoX/vignettes/example_UC_data/fibroblast_met.csv&quot;
output_path = &quot;./output/&quot;
# read in data
expression_mat = pd.read_csv(expression_file, index_col=0).T
meta = pd.read_csv(meta_file, index_col=0)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>print(expression_mat.shape)
print(meta.shape)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
(14822, 5494)
(14822, 10)
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>meta.head()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample</th>
      <th>disease</th>
      <th>cell_type</th>
      <th>cluster</th>
      <th>nGene</th>
      <th>nUMI</th>
      <th>percent_mito</th>
      <th>fibroblast_clusters</th>
      <th>UMAP1</th>
      <th>UMAP2</th>
    </tr>
    <tr>
      <th>cell</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>N7.LPA.AAACATACCATTTC</th>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>WNT2B+ Fos-lo 1</td>
      <td>550.0</td>
      <td>1118.0</td>
      <td>0.029517</td>
      <td>WNT2B</td>
      <td>9.517501</td>
      <td>2.068284</td>
    </tr>
    <tr>
      <th>N7.LPA.AAACGCTGTATCGG</th>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>Myofibroblasts</td>
      <td>636.0</td>
      <td>1312.0</td>
      <td>0.025915</td>
      <td>Myofibroblasts</td>
      <td>-0.044317</td>
      <td>-4.302498</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGACGAGACAGG</th>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>WNT2B+ Fos-hi</td>
      <td>1030.0</td>
      <td>2594.0</td>
      <td>0.021982</td>
      <td>WNT2B</td>
      <td>9.716670</td>
      <td>0.810530</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGATCTAGTCTG</th>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>WNT2B+ Fos-hi</td>
      <td>707.0</td>
      <td>1769.0</td>
      <td>0.019220</td>
      <td>WNT2B</td>
      <td>-4.460246</td>
      <td>6.526924</td>
    </tr>
    <tr>
      <th>N7.LPA.AAATACTGCTGGAT</th>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>WNT2B+ Fos-hi</td>
      <td>786.0</td>
      <td>1921.0</td>
      <td>0.021365</td>
      <td>WNT2B</td>
      <td>-5.412079</td>
      <td>6.846126</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
</section>
<section id="Step-2:-Preprocess-data---generate-latent-dimensions-and-configure-input-for-CellPhenoX-(include-covariates-and-identify-target-column)">
<h2>Step 2: Preprocess data - generate latent dimensions and configure input for CellPhenoX (include covariates and identify target column)<a class="headerlink" href="#Step-2:-Preprocess-data---generate-latent-dimensions-and-configure-input-for-CellPhenoX-(include-covariates-and-identify-target-column)" title="Link to this heading">¶</a></h2>
<p>First, we generate the NAM to summarize the cell co-abundance. CellPhenoX offers a wrapper around the Python <a class="reference external" href="https://github.com/immunogenomics/cna">CNA</a> package functions to accomplish this, <code class="docutils literal notranslate"><span class="pre">neighborhoodAbundanceMatrix</span></code>.</p>
<p>Parameters:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">expression_mat</span></code> - the single-cell omics feature matrix (with cells as rows and features as columns)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">meta</span></code> - corresponding meta data where columns are categories</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sampleid</span></code> - the name of the column in meta that has the individual/subject identification</p></li>
</ul>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>#nam = neighborhoodAbundanceMatrix(expression_mat=expression_mat, meta_data=meta, sampleid=&quot;sample&quot;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>nam = pd.read_csv(&quot;/Users/zhanglab_mac2/Library/CloudStorage/OneDrive-TheUniversityofColoradoDenver/Zhang_Lab/Research/shap/cell_neighborhood_abundance/NAM_cellXnbhd.csv&quot;, index_col=0)
nam.head()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>N7.LPA.AAACATACCATTTC</th>
      <th>N7.LPA.AAACGCTGTATCGG</th>
      <th>N7.LPA.AAAGACGAGACAGG</th>
      <th>N7.LPA.AAAGATCTAGTCTG</th>
      <th>N7.LPA.AAATACTGCTGGAT</th>
      <th>N7.LPA.AAATGGGAAAGTAG</th>
      <th>N7.LPA.AAATGTTGGTTACG</th>
      <th>N7.LPA.AACACGTGGAGAGC</th>
      <th>N7.LPA.AACCAGTGATCGTG</th>
      <th>N7.LPA.AACCGATGGAGAGC</th>
      <th>...</th>
      <th>N110.LPB.TTCCCAGTCAACGCTA</th>
      <th>N110.LPB.TTCTCAAAGTGCGTGA</th>
      <th>N110.LPB.TTCTCAATCCGTTGTC</th>
      <th>N110.LPB.TTCTTAGCAGCTATTG</th>
      <th>N110.LPB.TTGAACGCAATGACCT</th>
      <th>N110.LPB.TTGAACGGTCTGATTG</th>
      <th>N110.LPB.TTGGAACAGAGCTTCT</th>
      <th>N110.LPB.TTGGAACCAATCTACG</th>
      <th>N110.LPB.TTGTAGGCAGAGTGTG</th>
      <th>N110.LPB.TTTCCTCCATGCAATC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>N7.LPA.AAACATACCATTTC</th>
      <td>0.000073</td>
      <td>0.000026</td>
      <td>0.000053</td>
      <td>0.000046</td>
      <td>0.000037</td>
      <td>0.000050</td>
      <td>0.000053</td>
      <td>0.000044</td>
      <td>0.000061</td>
      <td>0.000050</td>
      <td>...</td>
      <td>0.000023</td>
      <td>0.000025</td>
      <td>0.000032</td>
      <td>0.000019</td>
      <td>0.000016</td>
      <td>0.000034</td>
      <td>0.000019</td>
      <td>0.000024</td>
      <td>0.000029</td>
      <td>0.000025</td>
    </tr>
    <tr>
      <th>N7.LPA.AAACGCTGTATCGG</th>
      <td>0.000026</td>
      <td>0.000097</td>
      <td>0.000034</td>
      <td>0.000026</td>
      <td>0.000024</td>
      <td>0.000036</td>
      <td>0.000037</td>
      <td>0.000027</td>
      <td>0.000030</td>
      <td>0.000033</td>
      <td>...</td>
      <td>0.000016</td>
      <td>0.000018</td>
      <td>0.000023</td>
      <td>0.000043</td>
      <td>0.000049</td>
      <td>0.000025</td>
      <td>0.000050</td>
      <td>0.000035</td>
      <td>0.000040</td>
      <td>0.000018</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGACGAGACAGG</th>
      <td>0.000053</td>
      <td>0.000034</td>
      <td>0.000078</td>
      <td>0.000049</td>
      <td>0.000037</td>
      <td>0.000053</td>
      <td>0.000047</td>
      <td>0.000047</td>
      <td>0.000055</td>
      <td>0.000049</td>
      <td>...</td>
      <td>0.000023</td>
      <td>0.000025</td>
      <td>0.000032</td>
      <td>0.000026</td>
      <td>0.000023</td>
      <td>0.000035</td>
      <td>0.000026</td>
      <td>0.000031</td>
      <td>0.000035</td>
      <td>0.000026</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGATCTAGTCTG</th>
      <td>0.000046</td>
      <td>0.000026</td>
      <td>0.000049</td>
      <td>0.000064</td>
      <td>0.000042</td>
      <td>0.000042</td>
      <td>0.000033</td>
      <td>0.000048</td>
      <td>0.000041</td>
      <td>0.000039</td>
      <td>...</td>
      <td>0.000041</td>
      <td>0.000043</td>
      <td>0.000051</td>
      <td>0.000023</td>
      <td>0.000018</td>
      <td>0.000053</td>
      <td>0.000023</td>
      <td>0.000033</td>
      <td>0.000030</td>
      <td>0.000042</td>
    </tr>
    <tr>
      <th>N7.LPA.AAATACTGCTGGAT</th>
      <td>0.000037</td>
      <td>0.000024</td>
      <td>0.000037</td>
      <td>0.000042</td>
      <td>0.000069</td>
      <td>0.000037</td>
      <td>0.000037</td>
      <td>0.000043</td>
      <td>0.000036</td>
      <td>0.000039</td>
      <td>...</td>
      <td>0.000045</td>
      <td>0.000046</td>
      <td>0.000055</td>
      <td>0.000023</td>
      <td>0.000018</td>
      <td>0.000054</td>
      <td>0.000026</td>
      <td>0.000034</td>
      <td>0.000029</td>
      <td>0.000046</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 14822 columns</p>
</div></div>
</div>
<p>CellPhenoX has two functions that will decompose the NAM, <code class="docutils literal notranslate"><span class="pre">nonnegativeMatrixFactorization</span></code> and <code class="docutils literal notranslate"><span class="pre">principalComponentAnalysis</span></code>.</p>
<p><em>Note</em>: if using a different dimensionality reduction technique, ensure that the format of the latent embeddings are in the following format for the preprocessing step (cells as rows, latent embeddings as columns).</p>
<p><code class="docutils literal notranslate"><span class="pre">nonnegativeMatrixFactorization</span></code> uses NMF to generate the ranks from the NAM.</p>
<p>Parameters:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">numberOfComponents</span></code> - the number of ranks (latent embeddings) to learn. Default = -1; if the number of components is not specified, then we select the number of components by performing NMF on a range of k values and selecting the one with the highest silhouette score.</p>
<ul>
<li><p><code class="docutils literal notranslate"><span class="pre">min_k</span></code> - minimum for the range</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">max_k</span></code> - maximum for the range</p></li>
</ul>
</li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># get the latent dimensions using NMF
latent_features = pyCellPhenoX.nonnegativeMatrixFactorization(nam, numberOfComponents=4, min_k=3, max_k=5)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
inside the NMF function
</pre></div></div>
</div>
<p>Alternatively, we can perform PCA with <code class="docutils literal notranslate"><span class="pre">principalComponentAnalysis</span></code> and use the principal components as the latent embeddings.</p>
<p>Paramters:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">var</span></code> - the desired proportion of variance explained. This will be used to select the number of principal components.</p></li>
</ul>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>proportion_var_explained = 0.9
latent_features = pyCellPhenoX.principalComponentAnalysis(nam, var=proportion_var_explained)
</pre></div>
</div>
</div>
<p>Now that we have the latent dimensions, the <code class="docutils literal notranslate"><span class="pre">preprocessing</span></code> function will add desired covariates to the model predictor dataframe, <code class="docutils literal notranslate"><span class="pre">X</span></code>, label encode any non-numerical columns to be used in the classification model, and optionally perform subsampling.</p>
<p>Parameters:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">sub_samp</span></code> - boolean flag, do you want to subsample your data</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">subset_percentage</span></code> - if <code class="docutils literal notranslate"><span class="pre">sub_samp</span></code> is True, specify the desired proportion of the dataset</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">bal_col</span></code> - name of columns to use to balance the subsampling</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">target</span></code> - name of the column in the meta data for the target/outcome variable for the classification model</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">covariates</span></code> - list of column names for covariates that you want to include as features in the classification model</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>X,y = pyCellPhenoX.preprocessing(latent_features, meta, sub_samp=False, bal_col=[&quot;subject_id&quot;, &quot;cell_type&quot;, &quot;disease&quot;], subset_percentage=0.25 , target=&quot;disease&quot;, covariates=[], interaction_covs=[])
X.head()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>LD_1</th>
      <th>LD_2</th>
      <th>LD_3</th>
      <th>LD_4</th>
    </tr>
    <tr>
      <th>cell</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>N7.LPA.AAACATACCATTTC</th>
      <td>0.009308</td>
      <td>0.000000</td>
      <td>0.001079</td>
      <td>0.000979</td>
    </tr>
    <tr>
      <th>N7.LPA.AAACGCTGTATCGG</th>
      <td>0.003498</td>
      <td>0.006898</td>
      <td>0.000760</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGACGAGACAGG</th>
      <td>0.009174</td>
      <td>0.000690</td>
      <td>0.001217</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGATCTAGTCTG</th>
      <td>0.006249</td>
      <td>0.000114</td>
      <td>0.002191</td>
      <td>0.000024</td>
    </tr>
    <tr>
      <th>N7.LPA.AAATACTGCTGGAT</th>
      <td>0.004989</td>
      <td>0.000571</td>
      <td>0.002303</td>
      <td>0.000273</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
</section>
<section id="Step-3:-Run-CellPhenoX---train-the-classification-model-and-calculate-the-CellPhenoX-Interpretable-Score">
<h2>Step 3: Run CellPhenoX - train the classification model and calculate the CellPhenoX Interpretable Score<a class="headerlink" href="#Step-3:-Run-CellPhenoX---train-the-classification-model-and-calculate-the-CellPhenoX-Interpretable-Score" title="Link to this heading">¶</a></h2>
<p>We start by instantiating the CellPhenoX object with <code class="docutils literal notranslate"><span class="pre">CellPhenoX</span></code>, specifying the following attributes:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">outer_num_splits</span></code> - the number of cross validation folds.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">inner_num_splits</span></code> - the number of cross validation folds for the hyperparameter grid search.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CV_repeats</span></code> - the number of cross validation repeats to perform. This ensures that the SHAP values calculated are robust to the randomization of the folds; the final SHAP values for each cell for each feature will be the average across CV repeats.</p></li>
</ul>
<p>The function for running this step is <code class="docutils literal notranslate"><span class="pre">model_training_shap_val</span></code>. You can pass the path to a specific output path for the produced plots.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># create CellPhenoX object
cellpx_obj = pyCellPhenoX.CellPhenoX(X, y, CV_repeats=1, outer_num_splits=3, inner_num_splits=2)
# and then train the classification model
cellpx_obj.model_training_shap_val(outpath = &quot;/Users/zhanglab_mac2/Library/CloudStorage/OneDrive-TheUniversityofColoradoDenver/Zhang_Lab/Research/shap/xcell/xcell_tutorial/xcell_results/ulcerative_colitis&quot;)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
entering CV loop

------------ CV Repeat number: 1

------ Fold Number: 1
--- Accuracy:  0.7589556769884639
--- Validation Accuracy:  0.8371269600404654  - Validation AUROC:  0.906510750194665  - Val AUPRC:  0.9319760696879232

------ Fold Number: 2
--- Accuracy:  0.7623962760574783
--- Validation Accuracy:  0.8730399595346484  - Validation AUROC:  0.9506476072147357  - Val AUPRC:  0.9688343084059845

------ Fold Number: 3
--- Accuracy:  0.7562753036437248
--- Validation Accuracy:  0.8148710166919575  - Validation AUROC:  0.8948686063679958  - Val AUPRC:  0.9241212208127081
best model precision-recall score = 0.9688
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/Users/zhanglab_mac2/Library/CloudStorage/OneDrive-TheUniversityofColoradoDenver/Zhang_Lab/Research/shap/.conda/lib/python3.11/site-packages/shap/plots/_beeswarm.py:699: UserWarning: No data for colormapping provided via &#39;c&#39;. Parameters &#39;vmin&#39;, &#39;vmax&#39; will be ignored
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/vignettes_test_vignette_20_2.png" src="../_images/vignettes_test_vignette_20_2.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/vignettes_test_vignette_20_3.png" src="../_images/vignettes_test_vignette_20_3.png" />
</div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">.shap_df</span></code> attribute will hold the SHAP values for the individual features in the model (these columns will have the names of the features and “_shap”). The <code class="docutils literal notranslate"><span class="pre">interpretable_score</span></code> column holds the CellPhenoX Interpretable Score.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>shap_df = cellpx_obj.shap_df
shap_df.head()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>LD_1_shap</th>
      <th>LD_2_shap</th>
      <th>LD_3_shap</th>
      <th>LD_4_shap</th>
      <th>interpretable_score</th>
    </tr>
    <tr>
      <th>cell</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>N7.LPA.AAACATACCATTTC</th>
      <td>-0.041487</td>
      <td>-0.277409</td>
      <td>-0.122060</td>
      <td>0.146939</td>
      <td>-0.294018</td>
    </tr>
    <tr>
      <th>N7.LPA.AAACGCTGTATCGG</th>
      <td>-0.093365</td>
      <td>0.033859</td>
      <td>-0.033869</td>
      <td>-0.198160</td>
      <td>-0.291535</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGACGAGACAGG</th>
      <td>-0.057741</td>
      <td>0.022331</td>
      <td>-0.125413</td>
      <td>-0.143444</td>
      <td>-0.304266</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGATCTAGTCTG</th>
      <td>-0.004691</td>
      <td>-0.132704</td>
      <td>-0.049079</td>
      <td>-0.120268</td>
      <td>-0.306743</td>
    </tr>
    <tr>
      <th>N7.LPA.AAATACTGCTGGAT</th>
      <td>0.011309</td>
      <td>-0.081440</td>
      <td>-0.104100</td>
      <td>0.068575</td>
      <td>-0.105656</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
</section>
<section id="Step-4:-Visualization---generate-manuscript-plots">
<h2>Step 4: Visualization - generate manuscript plots<a class="headerlink" href="#Step-4:-Visualization---generate-manuscript-plots" title="Link to this heading">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># VISUALIZATION
from pyCellPhenoX.plot_interpretablescore_umap import plot_interpretablescore_umap
from pyCellPhenoX.plot_interpretablescore_boxplot import plot_interpretablescore_boxplot
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>shap_df = shap_df.set_index(meta.index)
results_df = pd.concat([shap_df, meta], axis=1)
results_df.head()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>LD_1_shap</th>
      <th>LD_2_shap</th>
      <th>LD_3_shap</th>
      <th>LD_4_shap</th>
      <th>interpretable_score</th>
      <th>sample</th>
      <th>disease</th>
      <th>cell_type</th>
      <th>cluster</th>
      <th>nGene</th>
      <th>nUMI</th>
      <th>percent_mito</th>
      <th>fibroblast_clusters</th>
      <th>UMAP1</th>
      <th>UMAP2</th>
    </tr>
    <tr>
      <th>cell</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>N7.LPA.AAACATACCATTTC</th>
      <td>-0.041487</td>
      <td>-0.277409</td>
      <td>-0.122060</td>
      <td>0.146939</td>
      <td>-0.294018</td>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>WNT2B+ Fos-lo 1</td>
      <td>550.0</td>
      <td>1118.0</td>
      <td>0.029517</td>
      <td>WNT2B</td>
      <td>9.517501</td>
      <td>2.068284</td>
    </tr>
    <tr>
      <th>N7.LPA.AAACGCTGTATCGG</th>
      <td>-0.093365</td>
      <td>0.033859</td>
      <td>-0.033869</td>
      <td>-0.198160</td>
      <td>-0.291535</td>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>Myofibroblasts</td>
      <td>636.0</td>
      <td>1312.0</td>
      <td>0.025915</td>
      <td>Myofibroblasts</td>
      <td>-0.044317</td>
      <td>-4.302498</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGACGAGACAGG</th>
      <td>-0.057741</td>
      <td>0.022331</td>
      <td>-0.125413</td>
      <td>-0.143444</td>
      <td>-0.304266</td>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>WNT2B+ Fos-hi</td>
      <td>1030.0</td>
      <td>2594.0</td>
      <td>0.021982</td>
      <td>WNT2B</td>
      <td>9.716670</td>
      <td>0.810530</td>
    </tr>
    <tr>
      <th>N7.LPA.AAAGATCTAGTCTG</th>
      <td>-0.004691</td>
      <td>-0.132704</td>
      <td>-0.049079</td>
      <td>-0.120268</td>
      <td>-0.306743</td>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>WNT2B+ Fos-hi</td>
      <td>707.0</td>
      <td>1769.0</td>
      <td>0.019220</td>
      <td>WNT2B</td>
      <td>-4.460246</td>
      <td>6.526924</td>
    </tr>
    <tr>
      <th>N7.LPA.AAATACTGCTGGAT</th>
      <td>0.011309</td>
      <td>-0.081440</td>
      <td>-0.104100</td>
      <td>0.068575</td>
      <td>-0.105656</td>
      <td>N7</td>
      <td>Non-inflamed</td>
      <td>LP</td>
      <td>WNT2B+ Fos-hi</td>
      <td>786.0</td>
      <td>1921.0</td>
      <td>0.021365</td>
      <td>WNT2B</td>
      <td>-5.412079</td>
      <td>6.846126</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>We present the step-by-step workflow for running CellPhenoX to identify specific fibroblasts with predictable capability in distinguishing inflamed from non-inflamed colon.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>b = plot_interpretablescore_boxplot(data=results_df, x=&quot;fibroblast_clusters&quot;, y=&#39;interpretable_score&#39;)
print(b)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/vignettes_test_vignette_27_0.png" class="no-scaled-link" src="../_images/vignettes_test_vignette_27_0.png" style="width: 640px; height: 480px;" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

</pre></div></div>
</div>
<p>CellPhenoX Interpretable Score on the UMAP &amp; cell type labeled accordingly. This further confirmed that the identified inflammatory fibroblasts consistently show the highest scores, with a clear gradient extending toward a subset of WNT2B+ fibroblasts.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>c, s = plot_interpretablescore_umap(data=results_df, x=&quot;UMAP1&quot;, y=&quot;UMAP2&quot;, cell_type=&quot;fibroblast_clusters&quot;, score=&quot;interpretable_score&quot;)
print(c)
print(s)
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/vignettes_test_vignette_29_0.png" class="no-scaled-link" src="../_images/vignettes_test_vignette_29_0.png" style="width: 640px; height: 480px;" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/vignettes_test_vignette_29_2.png" class="no-scaled-link" src="../_images/vignettes_test_vignette_29_2.png" style="width: 640px; height: 480px;" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

</pre></div></div>
</div>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          
          
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024, Fan Zhang, Jade Young, Jun Inamo, Revanth Krishna, Zachary Caterer
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Applying CellPhenoX to Ulcerative Colitis Fibroblasts (single-cell transcriptomics)</a><ul>
<li><a class="reference internal" href="#Goal:-Identify-subtle,-fibroblast-specific-phenotypic-changes-associated-with-differing-levels-of-inflammation">Goal: Identify subtle, fibroblast-specific phenotypic changes associated with differing levels of inflammation</a></li>
<li><a class="reference internal" href="#Importing-Dependencies">Importing Dependencies</a></li>
<li><a class="reference internal" href="#Step-1:-Import-data">Step 1: Import data</a></li>
<li><a class="reference internal" href="#Step-2:-Preprocess-data---generate-latent-dimensions-and-configure-input-for-CellPhenoX-(include-covariates-and-identify-target-column)">Step 2: Preprocess data - generate latent dimensions and configure input for CellPhenoX (include covariates and identify target column)</a></li>
<li><a class="reference internal" href="#Step-3:-Run-CellPhenoX---train-the-classification-model-and-calculate-the-CellPhenoX-Interpretable-Score">Step 3: Run CellPhenoX - train the classification model and calculate the CellPhenoX Interpretable Score</a></li>
<li><a class="reference internal" href="#Step-4:-Visualization---generate-manuscript-plots">Step 4: Visualization - generate manuscript plots</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="../_static/documentation_options.js?v=baaebd52"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/scripts/furo.js?v=5fa4622c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../_static/readthedocs.js"></script>
    </body>
</html>